### ISSUE TITLE
Scripted Trades Errors with stress test

### ISSUE THREAD

**Issue by rkapl123** [OpenSourceRisk/Engine#215]
Hi, I'm trying to do stresstesting with scripted trades and with version 1.11 (we used ORE+ up to now), I'm getting errors like

ALERT    [2024-Jan-25 18:42:33.024348]  (...ics\orea\engine\valuationengine.cpp:327) : StructuredMessage { "category": "Error", "group": "Trade", "message": "date = 2024-01-24, sample = 0, label = BASE: Error during script execution: invalid map<K, T> key at L5:39:45: IF StructuredGearing*(StructuredIndex1(StructuredFixingSchedule[i]) - StructuredIndex2(StructuredFixingSchedule[i])) > StructuredIndex3(StructuredFixingSchedule[i]) THEN ", "sub_fields": [ { "name": "exceptionType", "value": "ScenarioValuation" }, { "name": "tradeId", "value": "<someIdentifier>" }, { "name": "tradeType", "value": "ScriptedTrade" } ] }

It seems to fail at evaluating StructuredIndex1(StructuredFixingSchedule[i]), but this only happens with stress test and not with simulation (which uses AMC) or valuation. Is this a known problem with structured trades in context of stress testing/sensitivity?

-regards,
Roland

**Comment by pcaspers** [comment]
Hi Roland, could you post an example please, we will need to debug this.

**Comment by rkapl123** [comment]
Hi Peter, due to market data copyright (Refinitiv/LSEG) I sent you the example directly.

regards,
Roland

**Comment by rkapl123** [comment]
Dear Peter!

Thank you for checking this on the last release, I built it now for myself and can confirm that the observed error is gone in https://github.com/OpenSourceRisk/Engine/tree/3aaa012bbbc3f8c02cdcc518ac4421261d63eedc (latest version as of today).
Also the problem with AMC purpose scripts unable to handle fixings earlier than the current AMC_SimDates node seems to be gone now, this required following workaround:
```
IF FundingAccrualSchedule[i + 1] > _AMC_SimDates[j] THEN 
    <code calculating coupon here...>
END;
```
-regards,
Roland

**Comment by rkapl123** [comment]
Yes, this seemed to be the problem here. Does it only work with "None", or can I also use "Defer" or "Unregister", as "None" has quite a performance penalty (especially with sensitivity).

-regards,
Roland

**Comment by pcaspers** [comment]
Hi Roland, I would expect that “None” works best, the other modes were more
designed for exposure simulation. I’ll have to check this or would you have
an example demonstrating this (not using scripted trades of course)?

Roland Kapl ***@***.***> schrieb am Mo. 29. Juli 2024 um
19:59:
